src/fns/unconstrained_helpers.nr

```diff
@@ -72,6 +72,54 @@ pub(crate) unconstrained fn __validate_gt_remainder<let N: u32>(
     (result, carry_flags, borrow_flags)
 }
 
+pub(crate) unconstrained fn __cmp_remainder<let N: u32>(
+    lhs: [Field; N],
+    rhs: [Field; N],
+) -> (bool, [Field; N], [bool; N], [bool; N]) {
+    let mut a_u60: U60Repr<N, 2> = U60Repr::from(lhs);
+    let mut b_u60: U60Repr<N, 2> = U60Repr::from(rhs);
+    let mut temp_u60: U60Repr<N, 2> = U60Repr::one();
+    let underflow = b_u60.gte(a_u60);
+
+    if underflow {
+        temp_u60 = a_u60;
```

Comment:
wonder how bad this is in brillig @TomAFrench. should I do what I did in the constrained version and just duplicate the code?

Reply:
Could you not refactor this function to have a private implementation which accepts two `U60Repr`s?

```
pub(crate) unconstrained fn __cmp_remainder<let N: u32>(
    lhs: [Field; N],
    rhs: [Field; N],
) -> (bool, [Field; N], [bool; N], [bool; N]) {
    let mut a_u60: U60Repr<N, 2> = U60Repr::from(lhs);
    let mut b_u60: U60Repr<N, 2> = U60Repr::from(rhs);
    let mut temp_u60: U60Repr<N, 2> = U60Repr::one();
    let underflow = b_u60.gte(a_u60);

    if underflow {
        __cmp_remainder_priv(b_u60, a_u60);
    } else {
        __cmp_remainder_priv(a_u60, b_u60);
    }
```

Again, benchmarking would be way to make these decisions.

Reply:
yep I can make a private function that does the second part (computing carries etc). will do that now.

Reply:
did the refactoring. I think we can refactor it a lot here. these operations are done over and over in all the `with_flags` functions.